presto-common/src/main/java/com/facebook/presto/common/PrestoVersion.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PrestoVersion
+{
+    private static final Pattern VERSION_REGEX = Pattern.compile("^(0\\.([1-9][0-9]*))(\\.([1-9][0-9]*))?(\\.fb-([1-9][0-9]*))?(-([2][0-9]{7}\\.[0-2][0-9]{5}-[0-9]{1,}|SNAPSHOT))?$");
+    private int majorVersion;
+    private int minorVersion;
+    private String snapshotVersion;
+
+    public PrestoVersion(String version)
+    {
+        Matcher matcher = VERSION_REGEX.matcher(version);
+        this.majorVersion = Integer.parseInt(Optional.ofNullable(matcher.find() ? matcher.group(2) : null).orElse("0"));
+        this.minorVersion = Integer.parseInt(Optional.ofNullable(matcher.find() ? matcher.group(4) : null).orElse("0"));
+        this.snapshotVersion = Optional.ofNullable(matcher.find() ? matcher.group(8) : null).orElse("");
+    }
+    public boolean equalTo(PrestoVersion version)
+    {
+        return (this.majorVersion == version.majorVersion
+                && this.minorVersion == version.minorVersion
+                && (this.snapshotVersion.isEmpty()
+                || version.snapshotVersion.isEmpty()
+                || this.snapshotVersion.equals(version.snapshotVersion)
+                || this.snapshotVersion.equals("SNAPSHOT")
+                || version.snapshotVersion.equals("SNAPSHOT")));
+    }
+    public boolean lessThan(PrestoVersion version)
+    {
+        if (this.majorVersion != version.majorVersion) {
+            return this.majorVersion < version.majorVersion;
+        }
+        if (this.minorVersion != version.minorVersion) {
+            return this.minorVersion < version.minorVersion;
+        }
+
+        return (!(this.snapshotVersion.equals("SNAPSHOT") || version.snapshotVersion.equals("SNAPSHOT"))
+                || (this.snapshotVersion.compareTo(version.snapshotVersion) < 0));
```

Comment:
I guess the other way is to extract the timestamp from snapshot.
What I was thinking is, the snapshot string is always going to be timestamp based, the string comparision may work because the older PrestoVersion would have the smaller string value lexicographically. 
Am I thinking this wrong? Is there something that we could use instead that would make more sense?